index.js

```diff
@@ -843,34 +757,30 @@ module.exports = function (log, indexesPath) {
   }
 
   function filterIndex(op, filterCheck, cb) {
-    ensureIndexSync(op, () => {
```

Comment:
are all of these remove ensureIndexSync because executeOperation now makes sure that indexes are ready, so we don't need to check this everywhere?